Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -92,6 +92,100 @@ namespace ShaderManagementConsole
+                shaderItemList, &ShaderManagementConsoleRequestBus::Events::GetMaterialInstanceShaderItems, materialAssetIdList[i]);
+
+            for (auto& shaderItem : shaderItemList)
+            {
+                AZ::RPI::ShaderVariantId shaderVariantId = shaderItem.GetShaderVariantId();
+
+                if (!shaderVariantId.IsEmpty())
+                {
+                    if (statisticData.m_shaderVariantUsage.find(shaderVariantId) == statisticData.m_shaderVariantUsage.end())
+                    {
+                        // Varient not found
+                        statisticData.m_shaderVariantUsage[shaderVariantId].m_count = 1;
+                    }
+                    else
+                    {
+                        statisticData.m_shaderVariantUsage[shaderVariantId].m_count += 1;
+                    }
+
+                    AZ::RPI::ShaderOptionGroup shaderOptionGroup = shaderItem.GetShaderOptionGroup();
+                    statisticData.m_shaderVariantUsage[shaderVariantId].m_shaderOptionGroup = shaderOptionGroup;
+
+                    for (auto& shaderOptionDescriptor : shaderOptionGroup.GetShaderOptionDescriptors())
+                    {
+                        AZ::Name optionName = shaderOptionDescriptor.GetName();
+                        AZ::RPI::ShaderOptionValue optionValue = shaderOptionGroup.GetValue(optionName);
+
+                        if (!optionValue.IsValid())
+                        {
+                            continue;
+                        }
+                        AZ::Name valueName = shaderOptionDescriptor.GetValueName(optionValue);
+                        if (statisticData.m_shaderOptionUsage.find(optionName) == statisticData.m_shaderOptionUsage.end())
+                        {
+                            // Option Not found
+                            statisticData.m_shaderOptionUsage[optionName][valueName] = 1;
+                        }
+                        else
+                        {
+                            if (statisticData.m_shaderOptionUsage[optionName].find(valueName) ==
+                                statisticData.m_shaderOptionUsage[optionName].end())
+                            {
+                                // Value not found
+                                statisticData.m_shaderOptionUsage[optionName][valueName] = 1;
+                            }
+                            else
+                            {
+                                statisticData.m_shaderOptionUsage[optionName][valueName] += 1;
+                            }
+                        }
+                    }
+                }
+            }
+
+            progressDialog.setValue(i);
+
+            if (progressDialog.wasCanceled())
+            {
+                return;
+            }
+        }
+        progressDialog.close();
+
+        AddDocumentTab(AZ::Uuid::CreateRandom(), new ShaderManagementConsoleStatisticView(statisticData, this));
```

Comment:
Is this supposed to create a view of statistics for all materials in the entire project or just statistics for materials using options in the currently loaded shader variant list?

If this is supposed to be a global view, not directly connected with a document, then I would recommend not adding a dummy document to register the view with. It does not need to be tabbed with the other documents. Instead, you can just create a regular, free floating or docked window.

If this is supposed to focus on statistics relative to the open shader variant list document, then I would instead create and register a tab or stack widget with a table view and statistics view as child pages, and register that as the document view. That way everything related to the document is packaged together.